hyperspy/_signals/eds_tem.py

```diff
@@ -353,37 +381,166 @@ def quantification(self,
+        else:
+            toa = take_off_angle
+
+        #determining illumination area for cross sections quantification.
+        if method == 'cross_section':
+            if probe_area == 'auto':
+                parameters = self.metadata.Acquisition_instrument.TEM
+                if probe_area in parameters:
+                    area = parameters.TEM.probe_area
+                else:
+                    if (self.axes_manager.navigation_dimension > 2 and
+                            navigation_axes is None):
+                        raise ValueError("With `probe_area='auto' and "
+                                         "navigation dimension > 2, you need "
+                                         "to specify the `navigation_axes` "
+                                         "parameter.")
+                    scales = []
+                    navigation_axes = self.axes_manager.navigation_axes
+                    for axis in navigation_axes:
+                        scales.append(
+                            axis.convert_to_units('nm', inplace=False)[0])
+                    if len(scales) == 1:
+                        probe_area = scales[0] * scales[0]
+                    elif len(scales) == 2:
+                        probe_area = scales[0] * scales[1]
+                    if scales[0] == 1 or scales[1] == 1:
+                        warnings.warn('Please note your probe_area is set to '
+                                      'the default value of 1 nm². The '
+                                      'function will still run. However if '
+                                      '1 nm² is not correct, please read the '
+                                      'user documentations for how to set '
+                                      'this properly.')
+
         if method == 'CL':
-            composition.data = utils_eds.quantification_cliff_lorimer(
-                composition.data, kfactors=factors,
-                mask=navigation_mask) * 100.
+            int_stack = utils.stack(intensities)
+
+            comp_old = utils.stack(intensities)
+            comp_old.data = np.zeros_like(comp_old.data)
+
+            abs_corr = None # initial
+
+            while True:
+                results = utils_eds.quantification_cliff_lorimer(
+                    int_stack.data,
+                    kfactors=factors,
+                    absorption_correction=abs_corr)
+                composition.data = results * 100.
+                if absorption_correction != False:
+                    if thickness is not None:
+                        mass_thickness = intensities[0].deepcopy()
+                        mass_thickness.data = self.CL_get_mass_thickness(composition.split(),
+                                                                    thickness)
+                        mass_thickness.metadata.General.title = 'Mass thickness'
+                    else:
+                        warnings.warn('Thickness is required for absorption'\
```

Comment:
I've added the spaces but not sure what you need by the quotes not being necessary. Is this removing all the quote marks in lines 452-454 or just leaving first and last in?

Reply:
I suspect I meant the `\` character!
Generally, I prefer to avoid using the `\` character when it is not necessary (as in this case) to make the code more readable and avoid unexpected behaviour/mistake since `\` can be used to escape other characters in string. This is a bit of matter of taste, so up to you, but the following should work fine:
```python
                        warnings.warn('Thickness is required for absorption '
                                      'correction with K-Factor Method. Result '
                                      'will contain no correction for absorption.')
```

Use "k-factor" instead of "K-Factor"?

There is still a space missing in the first line.